Automattic/jetpack
# Extensions: Add `plugins` folder. Move `Sharing` into it.

This PR adds a new `plugins` folder into the `extensions` one. The compiling system will take a look at this folder too, making it possible to put block editor plugins there.

**Why?**, because Sharing is not a block, it's a plugin.

**Why?**, because it's registered by the [registerJetpackPlugin()](https://github.com/Automattic/jetpack/blob/aebf634358e382e103a10ab06776e3f332f5c4cb/projects/plugins/jetpack/extensions/shared/register-jetpack-plugin.js#L18) function, instead of the [registerJetpackBlock()](https://github.com/Automattic/jetpack/blob/aebf634358e382e103a10ab06776e3f332f5c4cb/projects/plugins/jetpack/extensions/shared/register-jetpack-block.js#L75).

**Why?** Why not.

Indeed, It moves the `sharing` (ex) block to the plugins in this same PR.

Fixes #20943

#### Changes proposed in this Pull Request:

* Add plugins extensions folder. Move sharing into it.

#### Jetpack product discussion


p1630592306386100?thread_ts=1630590452.384100&cid=C01VDMPHG69-slack-

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Test locally with a Jetpack
* Clean the compiled files in the Jetpack plugin

Into Jetpack plugin folder...

```bash
cd projects/plugins/jetpack
```

... clean built files.
```bash
rm -rf _inc/blocks
```

... and compile again
```bash
pnpm run build-extensions
```

* Confirm `sharing` plugin is working as usual
<img src="https://user-images.githubusercontent.com/77539/131915990-caff56e0-b26b-4249-9c3d-06b51040eded.png" width="400" />
